Data_Structures/queues/cpp-queues.cpp

```diff
@@ -0,0 +1,102 @@
+/**Author : Vivek Bhardwaj
+  *Written for : Yet_Another_Algorithms_Repository
+**/
+
+/**IMPLEMENTATION OF QUEUE CLASS IN CPP
+  *INSPIRED BY QUEUE CLASS PROVIDED BY STL
+**/
+
+#include<iostream>
+using namespace std;
+
+
+/**Node structure created for linked list implementation of queue.
+  *The structure consists of template ,thus queue can be made for diffrent datatypes
+**/ 
+template<class A>
+struct Node{
+    A data;
+    Node * link;
+};
+
+template<class A>
+class queue{
+    private :
+        Node<A> * front_ptr;                    //front_ptr points to the Node in front of queue
+        Node<A> * back_ptr;                     //back_ptr points to the Node in back of queue
+        long long size;                         //size stores the no. of elements currently in queue
+
+    public :
+
+    /** Constructor which assigns values to the private variables
+    **/
+        queue(){
+            front_ptr=nullptr;
+            back_ptr=nullptr;
+            size=0;
+        }
+    /** Push function for the queue 
+     *  it creates a node and inserts it in end of the linked list
+    **/
+        void push(A x){
+            Node<A> *temp=new Node<A>;
+            temp->data=x;
```

Comment:
Just a suggestion, instead of making Node as a structure, please make it as a class with 2 constructors
1. with data as a parameter
2. with data as well as link as the parameters.

To avoid explicitly assigning values.